lib/transport/net-transport.js

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var AbstractTransport = require('./abstract-transport.js');
+var net = require('net');
+var debug = require('debug')('amqp10:connection:nettransport');
+
+var NetTransport = function () {
+	AbstractTransport.call(this);
+	this._transport = null;
+};
+
+NetTransport.prototype = Object.create(AbstractTransport.prototype);
+NetTransport.prototype.constructor = NetTransport;
+
+NetTransport.prototype.register = function (client) {
```

Comment:
tbh I'm not really a fan of this indirection. I'd prefer to just straight up call `AmqpClient.registerTransport(require('some-transport-module'))`. Then again I'm not too too picky about it, just seems needless.